### ISSUE TITLE
time index is unsorted or non-unique with multiple ensemble member array

### ISSUE THREAD

**Issue by pochedls** [NCAR/intake-esm#160]
## Description

This may be the same as (or has the same underlying cause) as #159. The issue is that when I create a dataset from a catalog search that includes multiple ensemble members, I cannot retrieve a time slice (but I am able to slice in time when my catalog search is for individual ensemble members). 

## Example

Minimal working example (changing `rip1 = r15i1p1f1` results in working code / no error messsage):

```
import intake


rip1 = 'r1i1p1f1'
rip2 = 'r11i1p1f1'
var_ids = ['tas']
exp_ids = ['historical']
col = intake.open_esm_datastore("catalogs/glade-cmip6.json")
catx = col.search(activity_id="CMIP", experiment_id=exp_ids,
                  source_id="EC-Earth3", member_id=[rip1, rip2],
                  variable_id=var_ids, table_id="Amon")
dset_dictx = catx.to_dataset_dict(zarr_kwargs={'consolidated': True,
                                             'decode_times': True},
                                cdf_kwargs={'chunks': {},
                                            'decode_times': True})
key = list(dset_dictx.keys())[0]
ds = dset_dictx[key]
dsvar = ds.tas.isel(member_id=0)
dsvar = dsvar.sel(time=slice('1979-01-01', '2014-12-31'))
```
## Error

> 
> KeyError                                  Traceback (most recent call last)
> ~/cmip6hack-pacific-sst/problem.py in <module>
>      18 ds = dset_dictx[key]
>      19 dsvar = ds.tas.isel(member_id=0)
> ---> 20 dsvar = dsvar.sel(time=slice('1979-01-01', '2014-12-31'))
> 
> /glade/work/pochedls/miniconda3/lib/python3.7/site-packages/xarray/core/dataarray.py in sel(self, indexers, method, tolerance, drop, **indexers_kwargs)
>    1045             method=method,
>    1046             tolerance=tolerance,
> -> 1047             **indexers_kwargs
>    1048         )
>    1049         return self._from_temp_dataset(ds)
> 
> /glade/work/pochedls/miniconda3/lib/python3.7/site-packages/xarray/core/dataset.py in sel(self, indexers, method, tolerance, drop, **indexers_kwargs)
>    1998         indexers = either_dict_or_kwargs(indexers, indexers_kwargs, "sel")
>    1999         pos_indexers, new_indexes = remap_label_indexers(
> -> 2000             self, indexers=indexers, method=method, tolerance=tolerance
>    2001         )
>    2002         result = self.isel(indexers=pos_indexers, drop=drop)
> 
> /glade/work/pochedls/miniconda3/lib/python3.7/site-packages/xarray/core/coordinates.py in remap_label_indexers(obj, indexers, method, tolerance, **indexers_kwargs)
>     390 
>     391     pos_indexers, new_indexes = indexing.remap_label_indexers(
> --> 392         obj, v_indexers, method=method, tolerance=tolerance
>     393     )
>     394     # attach indexer's coordinate to pos_indexers
> 
> /glade/work/pochedls/miniconda3/lib/python3.7/site-packages/xarray/core/indexing.py in remap_label_indexers(data_obj, indexers, method, tolerance)
>     259             coords_dtype = data_obj.coords[dim].dtype
>     260             label = maybe_cast_to_coords_dtype(label, coords_dtype)
> --> 261             idxr, new_idx = convert_label_indexer(index, label, dim, method, tolerance)
>     262             pos_indexers[dim] = idxr
>     263             if new_idx is not None:
> 
> /glade/work/pochedls/miniconda3/lib/python3.7/site-packages/xarray/core/indexing.py in convert_label_indexer(index, label, index_name, method, tolerance)
>     131                 "cannot represent labeled-based slice indexer for "
>     132                 "dimension %r with a slice over integer positions; "
> --> 133                 "the index is unsorted or non-unique" % index_name
>     134             )
>     135 
> 
> KeyError: "cannot represent labeled-based slice indexer for dimension 'time' with a slice over integer positions; the index is unsorted or non-unique"

**Comment by andersy005** [comment]
@pochedls, this issue seems to be fixed in #171.

Here's a notebook with details:
https://gist.github.com/andersy005/502bfd2b793baa1649fda28332c923f1#file-intake-esm-issue-160-ipynb

Once #171 is merged, feel free to update your environment by installing intake-esm from master branch

```bash
python -m pip install git+https://github.com/NCAR/intake-esm.git
```

**Comment by pochedls** [comment]
Thank you @andersy005!

**Comment by andersy005** [comment]
You are welcome, @pochedls! I am going to close this issue for now. If you encounter any other brick walls, feel free to reopen it or open a new issue.

**Comment by seismo2112** [comment]
I tried the same command but it displayed this error (in the end). Can you please look out at it?


 python -m pip install git+https://github.com/NCAR/intake-esm.git
Collecting git+https://github.com/NCAR/intake-esm.git
  Cloning https://github.com/NCAR/intake-esm.git to /tmp/pip-req-build-5f00gvb6
  Running command git clone -q https://github.com/NCAR/intake-esm.git /tmp/pip-req-build-5f00gvb6
  Resolved https://github.com/NCAR/intake-esm.git to commit 88f285fba2adfb590bdcfc07a768cf2b96f0e2e3
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: requests>=2.24.0 in ./.conda/envs/seismo/lib/python3.7/site-packages (from intake-esm==2021.8.17.post89) (2.27.1)
Collecting zarr>=2.5
  Downloading zarr-2.12.0-py3-none-any.whl (185 kB)
     |████████████████████████████████| 185 kB 1.1 MB/s
Collecting intake>=0.6.5
  Downloading intake-0.6.5-py3-none-any.whl (1.5 MB)
     |████████████████████████████████| 1.5 MB 9.7 MB/s
Collecting fsspec>=2021.7.0
  Downloading fsspec-2022.7.1-py3-none-any.whl (141 kB)
     |████████████████████████████████| 141 kB 31.4 MB/s
Collecting pydantic>=1.8.2
  Downloading pydantic-1.9.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.1 MB)
     |████████████████████████████████| 11.1 MB 20.9 MB/s
Collecting fastprogress>=1.0.0
  Downloading fastprogress-1.0.3-py3-none-any.whl (12 kB)
Requirement already satisfied: xarray!=0.20.0,!=0.20.1,>=0.19 in ./.conda/envs/seismo/lib/python3.7/site-packages (from intake-esm==2021.8.17.post89) (0.20.2)
Requirement already satisfied: netCDF4>=1.5.5 in ./.conda/envs/seismo/lib/python3.7/site-packages (from intake-esm==2021.8.17.post89) (1.5.7)
Collecting xcollection
  Downloading xcollection-2021.12.23-py3-none-any.whl (11 kB)
Collecting dask[complete]>=2021.9
  Downloading dask-2022.2.0-py3-none-any.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 18.4 MB/s
ERROR: Package 'intake-esm' requires a different Python: 3.7.11 not in '>=3.8'

**Comment by seismo2112** [comment]
> @pochedls, this issue seems to be fixed in #171.
> 
> Here's a notebook with details: https://gist.github.com/andersy005/502bfd2b793baa1649fda28332c923f1#file-intake-esm-issue-160-ipynb
> 
> Once #171 is merged, feel free to update your environment by installing intake-esm from master branch
> 
> ```shell
> python -m pip install git+https://github.com/NCAR/intake-esm.git
> ```

I tried the same command but it displayed this error (in the end). Can you please look out at it?

python -m pip install git+https://github.com/NCAR/intake-esm.git
Collecting git+https://github.com/NCAR/intake-esm.git
Cloning https://github.com/NCAR/intake-esm.git to /tmp/pip-req-build-5f00gvb6
Running command git clone -q https://github.com/NCAR/intake-esm.git /tmp/pip-req-build-5f00gvb6
Resolved https://github.com/NCAR/intake-esm.git to commit https://github.com/intake/intake-esm/commit/88f285fba2adfb590bdcfc07a768cf2b96f0e2e3
Installing build dependencies ... done
Getting requirements to build wheel ... done
Preparing wheel metadata ... done
Requirement already satisfied: requests>=2.24.0 in ./.conda/envs/seismo/lib/python3.7/site-packages (from intake-esm==2021.8.17.post89) (2.27.1)
Collecting zarr>=2.5
Downloading zarr-2.12.0-py3-none-any.whl (185 kB)
|████████████████████████████████| 185 kB 1.1 MB/s
Collecting intake>=0.6.5
Downloading intake-0.6.5-py3-none-any.whl (1.5 MB)
|████████████████████████████████| 1.5 MB 9.7 MB/s
Collecting fsspec>=2021.7.0
Downloading fsspec-2022.7.1-py3-none-any.whl (141 kB)
|████████████████████████████████| 141 kB 31.4 MB/s
Collecting pydantic>=1.8.2
Downloading pydantic-1.9.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.1 MB)
|████████████████████████████████| 11.1 MB 20.9 MB/s
Collecting fastprogress>=1.0.0
Downloading fastprogress-1.0.3-py3-none-any.whl (12 kB)
Requirement already satisfied: xarray!=0.20.0,!=0.20.1,>=0.19 in ./.conda/envs/seismo/lib/python3.7/site-packages (from intake-esm==2021.8.17.post89) (0.20.2)
Requirement already satisfied: netCDF4>=1.5.5 in ./.conda/envs/seismo/lib/python3.7/site-packages (from intake-esm==2021.8.17.post89) (1.5.7)
Collecting xcollection
Downloading xcollection-2021.12.23-py3-none-any.whl (11 kB)
Collecting dask[complete]>=2021.9
Downloading dask-2022.2.0-py3-none-any.whl (1.1 MB)
|████████████████████████████████| 1.1 MB 18.4 MB/s
ERROR: Package 'intake-esm' requires a different Python: 3.7.11 not in '>=3.8'

**Comment by andersy005** [comment]
@pochedls, 

>ERROR: Package 'intake-esm' requires a different Python: 3.7.11 not in '>=3.8'
 
as the error points out, you need to have Python >=3.8 and it looks like you are using Python 3.7. If you don't mind upgrade, i recommend upgrading your python version 

```bash
conda install 'python=3.8'
```
